composer.json

```diff
@@ -34,11 +34,15 @@
     "psr/log":                      "~1.0"
   },
   "require-dev": {
+    "symfony/asset":          "~2.7|~3.0|~4.0",
```

Comment:
Packages from `"require-dev"` are not installed in projects. So, only packages from `"require"` are installed, when you install this bundle. Packages from `"require-dev"` are installed only during the development of this bundle.
Are you ok with that? Does it resolve your initial problem?

Reply:
> So, only packages from "require" are installed, when you install this bundle (dev and non-dev too).
No, they wiil be installed by default https://getcomposer.org/doc/03-cli.md#install-i

> Does it resolve your initial problem?
Yes

Reply:
> > So, only packages from "require" are installed, when you install this bundle (dev and non-dev too).
> 
> No, they wiil be installed by default https://getcomposer.org/doc/03-cli.md#install-i

https://getcomposer.org/doc/04-schema.md#require-dev - read here. `require-dev` are only in `root-only` mode. So, it gets in consideration only `require-dev` from package.json from your project and ignores from dependencies. 
With current change `"symfony/asset"` will be installed only when you clone this repository and run `composer install` inside of it.